File: android/http-core/src/main/java/com/caldremch/http/core/abs/ICommonRequestEventCallback.java
package com.caldremch.http.core.abs;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Created by Leon on 2022/10/30.
 */
public interface ICommonRequestEventCallback {
   default void onStart() {};

    default void onSuccess(@Nullable Object obj){};

   default void onError(@NotNull Throwable th, boolean showToast){};

   default   void onEnd(){};
}

File: android-http/src/main/java/com/caldremch/http/core/abs/ICommonRequestEventCallback.kt
package com.caldremch.http.core.abs


/**
 *
 * @author Caldremch
 *
 * @date 2020-01-09 17:56
 *
 * @email caldremch@163.com
 *
 * @describe 可以处理Observer所有声明周期方法, 通用错误处理
 *
 **/
interface ICommonRequestEventCallback {
    fun onStart() {}
    fun onSuccess(data: Any?) {}
    //用户处理错误
    fun onError(e: Throwable)
    fun onEnd() {}
}